jsonrpc/eth_endpoint.go

```diff
@@ -550,19 +590,39 @@ func (e *Eth) GetLogs(filterOptions *LogFilter) (interface{}, error) {
 	return result, nil
 }
 
-// GetBalance returns the account's balance at the referenced block
-func (e *Eth) GetBalance(
-	address types.Address,
-	number *BlockNumber,
-) (interface{}, error) {
-	if number == nil {
-		number, _ = createBlockNumberPointer("latest")
+// GetBalance returns the account's balance at the referenced block.
+func (e *Eth) GetBalance(address types.Address, filter interface{}) (interface{}, error) {
```

Comment:
This applies to all the endpoints that implement this EIP:
- Avoid using an interface in the param, it is easier if the dispatcher can do the type uncasting instead.
- The code for all the functions is exactly the same, do not copy paste it every time because that is going to break for sure, please abstract that into a function ```getHeaderFromBlockNumberOrHash()```.
- Is there any unit test coverage of this?

Reply:
Thanks for the advices 💯 
I've made updates according to them in https://github.com/0xPolygon/polygon-sdk/pull/284/commits/6810ace3fb31792d7a8dcf4a084464e83961e769